mne/time_frequency/tfr.py

```diff
@@ -878,15 +878,36 @@ def crop(self, tmin=None, tmax=None):
             Start time of selection in seconds.
         tmax : float | None
             End time of selection in seconds.
+        fmin : float | None
+            Lowest frequency of selection in Hz.
+
+            .. versionadded:: 0.18.0
+        fmax : float | None
+            Highest frequency of selection in Hz.
+
+            .. versionadded:: 0.18.0
 
         Returns
         -------
         inst : instance of AverageTFR
             The modified instance.
         """
-        mask = _time_mask(self.times, tmin, tmax, sfreq=self.info['sfreq'])
-        self.times = self.times[mask]
-        self.data = self.data[..., mask]
+        if tmin is not None or tmax is not None:
+            time_mask = _time_mask(self.times, tmin, tmax,
+                                   sfreq=self.info['sfreq'])
+
+        else:
+            time_mask = slice(None)
+
+        if fmin is not None or fmax is not None:
+            freq_mask = _freq_mask(self.freqs, sfreq=self.info['sfreq'],
+                                   fmin=fmin, fmax=fmax)
+        else:
+            freq_mask = slice(None)
+
+        self.times = self.times[time_mask]
+        self.freqs = self.freqs[freq_mask]
+        self.data = self.data[..., freq_mask, :][..., time_mask]
```

Comment:
this `self.data[..., freq_mask, :][..., time_mask]` will still make 2 copies in memory.
this will not (unless it does not work):
`self.data[..., freq_mask, time_mask]`

Reply:
> with the slice() then I promise I stop.

I think you forgot your promise :)

> this will not (unless it does not work):

It will not work if both are ndarrays. Based on how NumPy does things, they need to obey broadcasting:
```
>>> x = np.zeros((2, 3, 4))
>>> freq_mask, time_mask = np.array([1, 2]), np.array([0, 2, 3])
>>> x[:, freq_mask, time_mask]
IndexError: shape mismatch: indexing arrays could not be broadcast together with shapes (2,) (3,)
```
So what we really need is to create a new dim on `freq_mask`: 
```
>>> x[:, freq_mask[:, np.newaxis], time_mask].shape
```
But the `newaxis` should be used only in the case that *both* are `ndarray`, because if `time_mask` is a slice and you still `np.newaxis` the `freq_mask` you get:
```
>>> x[:, freq_mask[:, np.newaxis], slice(None)].shape
(2, 2, 1, 4)
```

Reply:
ok you win !

Reply:
@larsoner @agramfort 
Yes, that was my problem too while trying to implement it:
https://github.com/mne-tools/mne-python/pull/6080#discussion_r270381268

Reply:
We can make `time_mask` and `freq_mask` be always a slice.